packages/paste-website/src/components/shortcodes/live-preview/index.tsx

```diff
@@ -1,64 +1,113 @@
 // https://github.com/FormidableLabs/react-live
 import * as React from 'react';
-import {Box} from '@twilio-paste/box';
 import {LiveProvider, LiveEditor, LiveError, LivePreview as ReactLivePreview} from 'react-live';
+import {Box} from '@twilio-paste/box';
+import {Button} from '@twilio-paste/button';
 import {useTheme} from '@twilio-paste/theme';
+import {useClipboard} from '@twilio-paste/clipboard-copy-library';
+import {ShowIcon} from '@twilio-paste/icons/esm/ShowIcon';
+import {HideIcon} from '@twilio-paste/icons/esm/HideIcon';
+import {CopyIcon} from '@twilio-paste/icons/esm/CopyIcon';
 import {CodeblockTheme} from './theme';
+import {CodeBlockOverlayShadow} from './CodeBlockOverlayShadow';
 import type {Language} from '../../codeblock';
 
-interface CodeblockProps {
+interface LivePreviewProps {
   children: string;
   scope: {[key: string]: any};
   language?: Language;
-  disabled?: boolean;
-  noInline?: boolean;
 }
 
-// FIXME use tokens for theme and LiveEditor
-const LivePreview: React.FC<CodeblockProps> = ({
-  children,
-  language = 'jsx',
-  scope,
-  disabled = false,
-  noInline = false,
-}) => {
+const LivePreview: React.FC<LivePreviewProps> = ({children, language = 'jsx', scope}) => {
+  const [viewCode, setViewCode] = React.useState(false);
+  const clipboard = useClipboard({copiedTimeout: 1200});
   const pasteTheme = useTheme();
+
+  const handleCopy = React.useCallback(() => {
```

Comment:
For my own understanding, why is it useful to do useCallback() here specifically?

Reply:
The library this uses recommends that approach.